src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/TagBuilder.cs

```diff
@@ -77,16 +77,46 @@ public static string CreateSanitizedId(string name, [NotNull] string invalidChar
                 return string.Empty;
             }
 
+            // If there are no invalid characters in the string, then we don't have to create the buffer.
+            var i = 0;
+            for (i = 1; i < name.Length; i++)
+            {
+                if (!Html401IdUtil.IsValidIdCharacter(name[i]))
+                {
+                    break;
+                }
+            }
+
             var firstChar = name[0];
-            if (!Html401IdUtil.IsAsciiLetter(firstChar))
+            var startsWithAsciiLetter = Html401IdUtil.IsAsciiLetter(firstChar);
+            if (!startsWithAsciiLetter)
             {
                 // The first character must be a letter according to the HTML 4.01 specification.
                 firstChar = 'z';
             }
 
+            if (i == name.Length)
+            {
+                if (startsWithAsciiLetter)
+                {
+                    return name;
+                }
+                else
+                {
+                    return firstChar + name.Substring(1);
```

Comment:
Profiling helps. Need to see data. For instance, I assume you haven't measured, because this *optimization* would allocate more than letting it fall through to the string builder case.